### ISSUE TITLE
Context menu section items don't report their position with a screen reader

### ISSUE THREAD

**Issue by bdjeffyp** [microsoft/fluentui#12594]
### Environment Information

- **Package version(s)**: 7.105.11
- **Browser and OS versions**: Edge Version 81.0.416.45 (Official build) beta (64-bit), Windows 10 Pro Version 1909 OS build 18363.752

### Describe the issue:
When using Narrator with keyboard navigation to operate a context menu item with sections, the contextual menu won't announce how many items exist within sections. Sections are considered to be navigable items by the screen reader when determining how many items exist within the contextual menu, but then items within the section are not announced their position out of the total number in the section. A look at the elements within the Dev Tools shows that section items do include "aria-posinset" and "aria-setsize" attributes, but Narrator does not announce these values.

### Please provide a reproduction of the issue in a codepen:
https://fabricweb.z5.web.core.windows.net/oufr/7.105.11/#/examples/contextualmenu
Look at the section titled **ContextualMenu with section headers** and click on the "Click for ContextualMenu" button to display the menu with the Narrator on.

#### Actual behavior:
When navigating the items in the menu, the screen reader says as the user presses the down key to go through the options, "Actions New Menu item", "Delete Menu item", "Social Share Menu item", "Print Menu item", and so on. This does not provide the screen reader user any idea of how many items there are within the menu sections or the entire menu, requiring them to cycle through the whole menu to know how many options there are.

#### Expected behavior:
It should say "Actions New Menu item 1 of 2", "Delete Menu item 2 of 2", "Social Share Menu item 1 of 3", and so on. Or, it should use the total across all sections like this: "Actions New Menu item 1 of 6", "Delete Menu item 2 of 6", and so on.

### Documentation describing expected behavior
https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html

**Comment by joschect** [comment]
@bdjeffyp Thanks for the submission. I can reproduce this behavior. I am a little confused because it appears that the contextualmenu section items have all of the information to be read out. They have the appropriate `aria-posinset` and `aria-setsize` (see screenshot). @micahgodbolt do you have any idea why this might not be getting read out?

![image](https://user-images.githubusercontent.com/20801176/78817507-169b2180-7988-11ea-8ac2-7abf151637d9.png)

**Comment by bdjeffyp** [comment]
Yep. That is what I experienced on my site. It is very weird why that would happen. I know some screen readers may have differences in how they work, but seeing it happen across multiple screen readers suggests that this is a problem beyond the screen readers themselves.

**Comment by joschect** [comment]
We'll get this marked as a bug and get it looked at!

**Comment by layershifter** [comment]
I checked this with @kolaps33 and we found out that the issue might be caused by wrong elements/roles.

Currently `ContextualMenu` renders following HTML (simplified 🐔 ):

```html
<div role="group">
  <ul>
    <li role="separator" aria-hidden="true"></li>
    <li role="presentation" class="ms-ContextualMenu-item item-205">
      <div>
        <div class="ms-ContextualMenu-linkContent linkContent-211">
          <span class="ms-ContextualMenu-itemText label-217">Actions</span>
        </div>
      </div>
    </li>
  </ul>
</div>
```

The issue that we see is that `role="group"` wraps `ul` (is equal to `role="list"`):

![An accessibility tree from Edge Devtools](https://user-images.githubusercontent.com/14183168/105046433-65bec880-5a69-11eb-91ee-e50764490a9c.png)

If we will change this to `presentation` and `group`:

```html
<div role="presentation">
  <ul role="group"></ul>
</div>
```

![An accessibility tree from Edge Devtools](https://user-images.githubusercontent.com/14183168/105046647-a6b6dd00-5a69-11eb-9e64-74aed0e0774a.png)

Then NVDA reads it properly:

```
Share
New  1 of 2
Delete  2 of 2
```

**Comment by smhigley** [comment]
I updated the ContextualMenu roles a bit in https://github.com/microsoft/fluentui/pull/17266, and it's working properly for me now with NVDA & Edge, Chrome, and FF. I'm not sure if it was that PR that fixed it, or an earlier one, but it should be OK now!
